librarymanagement/src/main/scala/sbt/librarymanagement/Artifact.scala

```diff
@@ -5,30 +5,47 @@ package sbt.librarymanagement
 
 import java.io.File
 import java.net.URL
-import sbt.serialization._
 
-final case class Artifact(name: String, `type`: String, extension: String, classifier: Option[String], configurations: Iterable[Configuration], url: Option[URL], extraAttributes: Map[String, String]) {
+abstract class ArtifactParent {
```

Comment:
Is this the mixin for the methods? If so the naming convention I think should be `ArtifactExtra`.
Sort of like `ProjectExtra` etc - https://github.com/sbt/sbt/blob/0.13/main/src/main/scala/sbt/Project.scala#L608